### PR TITLE
INGK-814 Add support to bits in PDOs

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

This PR adds support to bits in the PDO.

Fixes # INGK-814

### Type of change

- [x] Add bit type in for registers.
- [x] PDO module use bits internally and converts to bytes to send data to drive.

### Tests
- [x] Add new unit tests if it applies.
- [x] Run tests.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Build documentation locally to verify changes.
- [ ] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingenialink tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingenialink/utils/_utils.py [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 if TYPE_CHECKING:
     from ingenialink.servo import Servo
 
-import warnings
 import functools
-import ingenialogger
+import warnings
 import xml.etree.ElementTree as ET
 
+import ingenialogger
```
[the remark this answers is not in the document]
Comment (pzinemanas) [reply ingenialink/utils/_utils.py:15]: I'm using [isort](https://pypi.org/project/isort/) to automatically sort the imports.

File: tests/ethercat/test_pdo.py [hunk 2]
```diff
@@ -239,6 +244,7 @@ def test_servo_reset_pdos(connect_to_slave, create_pdo_map):
 
 
 @pytest.mark.ethercat
+@pytest.mark.skip
```
Comment (GullinBustin) [inline tests/ethercat/test_pdo.py:247]: Why do you skip?

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: tests/ethercat/test_pdo.py (hunk 2 above)
Comment (pzinemanas) [reply tests/ethercat/test_pdo.py]: It was mistake, I removed the mark: [27405b8](https://github.com/ingeniamc/ingenialink-python/pull/370/commits/27405b8a8349395b616558d793141ac57626fa44)

File: ingenialink/utils/_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GullinBustin) [reply ingenialink/utils/_utils.py]: But ingenialogger is a third party, it should be with the others, why did isort move it?

File: ingenialink/pdo.py [hunk 3]
```diff
@@ -69,7 +104,11 @@ def value(self) -> Union[int, float]:
         Returns:
             Register value.
         """
-        value = convert_bytes_to_dtype(self.raw_data, self.register.dtype)
+        value: Union[bool, int, float, str]
+        if self.register.dtype == REG_DTYPE.BOOL:
+            value = self.raw_data_bits.any()
```
Comment (polfeliu) [inline ingenialink/pdo.py:109]: Weird that any() is used for a bitarray of length 1, but I see not alternative

File: ingenialink/pdo.py [hunk 4]
```diff
@@ -83,7 +122,7 @@ def register_mapping(self) -> bytes:
 
         """
         index = self.register.idx
-        mapped_register = (index << 16) | (self.size * 8)
+        mapped_register = (index << 16) | self.size_bits
```
[the remark this answers is not in the document]
Comment (pzinemanas) [reply ingenialink/pdo.py:125]: I don't think so. So size_bits should be lower or equal to 16?

File: ingenialink/pdo.py [hunk 5]
```diff
@@ -232,29 +285,42 @@ class RPDOMap(PDOMap):
...
-                f"The length of the data array is incorrect. Expected {self.data_bytes_length},"
-                f" obtained {len(data_bytes)}"
+                "The length in bits of the data array is incorrect. Expected"
+                f" {self.data_length_bits}, obtained {len(data_bits)}"
             )
-        return data_bytes
+        return data_bits
+
+    def get_item_bytes(self) -> bytes:
+        """Return the concatenated items raw data to be sent to the drive (in bytes).
```
[the remark this answers is not in the document]
Comment (pzinemanas) [reply ingenialink/pdo.py]: Done!

File: ingenialink/pdo.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (polfeliu) [reply ingenialink/pdo.py:125]: Should be lower to 2^16 at least here, but ethernet frame itself is more restrictive so maybe it doesn't make any sense to check it

**Review by polfeliu — APPROVED** [review]

File: ingenialink/pdo.py [hunk 6]
```diff
@@ -203,17 +247,26 @@ def map_register_index(self, address: int) -> None:
...
 
         Returns:
-            Length of the map bytes.
+            Length of the map in bits.
         """
         length = 0
         for item in self.items:
-            length += item.size
+            length += item.size_bits
         return length
```
Comment (MartinALL91) [inline ingenialink/pdo.py:259]: You could do `return sum(item.size_bits for item in self.items)`

File: tests/ethercat/test_pdo.py [hunk 7]
```diff
@@ -96,7 +101,7 @@ def test_tpdo_item(open_dictionary):
     with pytest.raises(AttributeError):
         tpdo_item.value = 15
 
-    tpdo_item.raw_data = convert_dtype_to_bytes(15, REG_DTYPE.U16)
+    tpdo_item.raw_data_bytes = convert_dtype_to_bytes(15, REG_DTYPE.S32)
```
Comment (MartinALL91) [inline tests/ethercat/test_pdo.py:104]: You could use the dtype attribute

File: tests/ethercat/test_pdo.py [hunk 8]
```diff
@@ -52,14 +55,16 @@ def test_rpdo_item(open_dictionary):
     rpdo_item = RPDOMapItem(register)
 
     assert rpdo_item.register == register
-    assert rpdo_item.size == dtype_value[rpdo_item.register.dtype][0]
+    assert rpdo_item.size_bits == dtype_value[rpdo_item.register.dtype][0] * 8
```
Comment (MartinALL91) [inline tests/ethercat/test_pdo.py:58]: You could use the dtype_length_bits dictionary.

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**